packages/ember/tests/routing/router_service_test/currenturl_lifecycle_test.js

```diff
@@ -241,17 +245,17 @@ moduleFor(
             ['parent.index - parent.index', 'parent.child model', '/'],
             ['parent.index - parent.index', 'parent.loading activate', '/'],
             [
-              'parent.index - parent.index',
+              'parent.index - parent.loading',
```

Comment:
I'm concerned by this change, because this was the goal of https://github.com/emberjs/ember.js/pull/19450/ to make routerService.currentRouteName in sync with routerService.currentRoute.name

I think this check https://github.com/emberjs/ember.js/pull/19405/files#diff-02300d4f8ddd433fa4571addf26231e52255efd632d0b9972b02086584bf15ccR309 should be put at the end of the routeWillChange handler in the router

Reply:
@sly7-7 Ah, yes.  I wasn't exactly sure whether it was the test that needed fixing, but I understand it better now.  Changing the order of when the event is triggered on the router service seems to pass the tests. (with this test reverted to its original form)